deepchem/splits/splitters.py

```diff
@@ -763,6 +763,18 @@ class IndexSplitter(Splitter):
     training, the next `frac_valid` for validation, and the final `frac_test` for
     testing. This class may make sense to use your `Dataset` is already time
     ordered (for example).
+
+    Examples
+    --------
+    >>> n_samples = 100
+    >>> n_features = 10
+    >>> n_tasks = 10
+    >>> X = np.random.rand(n_samples, n_features)
+    >>> y = np.random.rand(n_samples, n_tasks)
+    >>> splitter = dc.splits.IndexSplitter()
+    >>> dataset = dc.data.NumpyDataset(X, y)
+    >>> train_dataset, valid_dataset , test_dataset = splitter.split(dataset)
```

Comment:
A `splitter.split` method will return indices for split and not datasets. Instead, if we use `train_dataset, test_dataset = splitter.train_test_split(dataset)`, it will be more correct.

Also, the example should demonstrate how the splitter works. I would rewrite it something like this:
```
n_samples = 5
n_features = 2
X = np.random.rand(n_samples, n_features)
y = np.random.rand(n_samples)
splitter = dc.splits.IndexSplitter()
dataset = dc.data.NumpyDataset(X, y)

>>> train_dataset, test_dataset = splitter.train_test_split(dataset)

>>> print(train_dataset.ids)
[0, 1, 2]
>>> print (test_dataset.ids)
[3, 4]
```
The above example demonstrates how the splitter splits by index.
```

Reply:
Thank u for your correction, i have fixed that.

Reply:
I mistook split function for something else.
I fixed that in the latest commit.